modules/data.atmosphere/R/debias_met_regression.R

```diff
@@ -0,0 +1,761 @@
+##'                  met variables that have been naively gapfilled for certain time periods
+##' @param pair.anoms - logical stating whether anomalies from the same year should be matched or not
+##' @param pair.ens - logical stating whether ensembles from train and source data need to be paired together 
+##'                   (for uncertainty propogation)
+##' @param uncert.prop - method for error propogation if only 1 ensemble member; options=c(random, mean); *Not Implemented yet
+##' @param resids - logical stating whether to pass on residual data or not *Not implemented yet
+##' @param seed - specify seed so that random draws can be reproduced
+##' @param outfolder - directory where the data should go
+##' @param yrs.save - what years from the source data should be saved; if NULL all years of the source data will be saved
+##' @param ens.name - what is the name that should be attached to the debiased ensemble
+##' @param ens.mems - what labels/numbers to attach to the ensemble members so we can gradually build bigger ensembles 
+##'                   without having to do do giant runs at once; if NULL will be numbered 1:n.ens
+##' @param lat.in - latitude of site
+##' @param lon.in - longitude of site
+##' @param save.diagnostics - logical; save diagnostic plots of output?
+##' @param path.diagnostics - path to where the diagnostic graphs should be saved
+##' @param parallel - (experimental) logical stating whether to run temporal_downscale_functions.R in parallel *Not Implemented yet
+##' @param n.cores - (experimental) how many cores to use in parallelization *Not implemented yet
+##' @param overwrite - overwrite existing files?
+##' @param verbose
+##' @export
+# -----------------------------------
+# Workflow
+# -----------------------------------
+# The general workflow is as follows: 
+# 1. read in & format data (coerce to daily format)
+# 2. set up the file structures for the output
+# 3. define the training window
+# 4. generate the bias-correction models to adjust the seasonal cycle
+# 5. Model the anomalies
+# 5. apply the climatology & anomaly models with covariance to generate a daily ensemble
+# 6. Save specified years to file
+# -----------------------------------
+
+#----------------------------------------------------------------------
+# Begin Function
+#----------------------------------------------------------------------
+
+
+debias.met.regression <- function(train.data, source.data, n.ens, vars.debias=NULL, CRUNCEP=FALSE,
+                                  pair.anoms = TRUE, pair.ens = FALSE, uncert.prop="mean", resids = FALSE, seed=Sys.Date(),
+                                  outfolder, yrs.save=NULL, ens.name, ens.mems=NULL, lat.in, lon.in,
+                                  save.diagnostics=TRUE, path.diagnostics=NULL,
+                                  parallel = FALSE, n.cores = NULL, overwrite = TRUE, verbose = FALSE) {
+  library(MASS)
```

Comment:
Same comment as before - on moving library calls into export, and explicitly labelling with ::